riptide-compression/src/main/java/org/zalando/riptide/compression/WrappingHttpOutputMessage.java

```diff
@@ -0,0 +1,43 @@
+package org.zalando.riptide.compression;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpOutputMessage;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.OutputStream;
+
+final class WrappingHttpOutputMessage implements HttpOutputMessage, AutoCloseable {
+
+    private final HttpOutputMessage message;
+    private final Compression.OutputStreamWrapper wrapper;
+    private OutputStream stream;
+
+    WrappingHttpOutputMessage(HttpOutputMessage message, Compression.OutputStreamWrapper wrapper) {
+        this.message = message;
+        this.wrapper = wrapper;
+    }
+
+    @Nonnull
+    @Override
+    public OutputStream getBody() throws IOException {
+        if (stream == null) {
+            stream = wrapper.wrap(message.getBody());
```

Comment:
Couldn't this be done in the constructor (+ `final`)?

Reply:
Some compressing stream wrappers (like `GZIPOutputStream`) write to the underlying stream in their constructor, therefore I thought it may make sense to delay this. Wdyt?

Reply:
In case there is nothing ever to write?

Reply:
If this is a scenario, yes. I wanted to be extra safe and "clean" (makes the code less clean though).